signed/sign.go

```diff
@@ -7,11 +7,19 @@ import (
 )
 
 func Sign(s *data.Signed, k *data.Key) {
+	id := k.ID()
+	signatures := make([]data.Signature, 0, len(s.Signatures))
```

Comment:
If there is no excess capacity, `append` returns a new slice. Otherwise it just uses the existing capacity that is greater than the length. The append below currently returns a new slice, but that is not optimal as it's an unnecessary memory allocation.